src/Storage/PathBuilder/PathBuilderTrait.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * @author Florian Krämer
+ * @copyright 2012 - 2015 Florian Krämer
+ * @license MIT
+ */
+namespace Burzum\FileStorage\Storage\PathBuilder;
+
+use Cake\Core\App;
+use RuntimeException;
+
+trait PathBuilderTrait {
```

Comment:
This duplicates the functionality [from the AbstractListener](https://github.com/burzum/cakephp-file-storage/blob/3.0-event-overhaul/src/Storage/Listener/AbstractListener.php#L311).

The code should be removed there and the trait used. I'll merge this but the duplicate code should be handled as well.